tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-interactive-inference-dashboard.html

```diff
@@ -4165,77 +4201,30 @@ <h2>Show similarity to selected datapoint</h2>
-              if (
-                this.stats[feature2].totalCount !=
-                this.examplesAndInferences.length
-              ) {
-                feature2Values = Object.assign({}, feature2Values, {
-                  undefined: '',
-                });
-              }
-            }
-
-            // For the selected feature, set up a dict of each feature value in
-            // the dataset to the threshold. Add this to a list (for display
-            // purposes) and create a map of feature value to entry in that list.
-            for (var key1 in feature1Values) {
-              if (feature1Values.hasOwnProperty(key1)) {
-                for (var key2 in feature2Values) {
-                  if (feature2Values.hasOwnProperty(key2)) {
-                    const feature1Value =
-                      key1 == 'undefined'
-                        ? undefined
-                        : this.stats[feature1].valueHash[key1].value;
-                    const feature2Value =
-                      key2 == 'undefined'
-                        ? undefined
-                        : this.stats[feature2].valueHash[key2].value;
-                    // Deep copy thresholds for each facet
-                    const modelThresholds = [];
-                    for (let i = 0; i < this.overallThresholds.length; i++) {
-                      modelThresholds.push({
-                        threshold: this.overallThresholds[i].threshold,
-                      });
-                    }
-                    const thresh = {
-                      value: feature1Value,
-                      value2: feature2Value,
-                      threshold: modelThresholds,
-                      opened: false,
-                    };
-                    thresholds.push(thresh);
-                    const mapKey = this.createCombinedValueString_(
-                      feature1Value,
-                      feature2Value
-                    );
-                    thresholdsMap[mapKey] = thresh;
-                  }
-                }
+          this.visdata.forEach(
+            function(item) {
+              const key1 = this.getSliceKey_(
+                item,
+                this.selectedBreakdownFeature
+              );
+              const key2 = this.getSliceKey_(
+                item,
+                this.selectedSecondBreakdownFeature
+              );
+              const key = this.createCombinedValueString_(key1, key2);
+              if (!(key in thresholdsMap)) {
+                thresholdsMap[key] = {
+                  value: key1,
+                  value2: key2,
+                  threshold: deepClone(this.overallThresholds),
+                  opened: false,
+                };
               }
-            }
-          }
+            }.bind(this)
+          );
+          const thresholds = Object.values(thresholdsMap);
+          this.set('selectedFeatureSort', 'Alphabetical');
```

Comment:
why reset the sort order when slices change? i can imagine wanting to keep viewing slices by something like false positive rate as i change slicing criteria.